resources/todomvc/big-dom-generator/src/tree-generator.js

```diff
@@ -8,26 +8,49 @@ const random = new LCG(DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR);
  * It starts with the minimum number of maximum-depth branches and randomly adds
  * children to the nodes in a breadth-first manner.
  * The weight parameters represent how many DOM nodes are generated for each type of node.
+ * @param {number} listWeight - The weight for the "list" node type. <ul></ul>
+ * @param {number} expandableItemWeight - The weight for the "expandableItem" node type. <li></li> with ChevronRight svg.
+ * @param {number} nonExpandableItemWeight - The weight for the "nonExpandableItem" node type. <li></li> TaskListIcon svg.
+ * @returns {Object} The generated tree structure. Example structure:
+ * {
+ *    type: "list",
+ *    children: [
+ *      {
+ *         type: "expandableItem",
+ *         children: [
+ *           {
+ *             type: "list",
+ *             children: [
+ *               {
+ *                 type: "nonExpandableItem",
+ *                 children: []
+ *               }
+ *             ]
+ *           }
+ *         ]
+ *      }
+ *    ]
+ * }
  **/
-export const generateTreeHead = ({ listWeight, openItemWeight, closedItemWeight }) => {
+export const generateTreeHead = ({ listWeight, expandableItemWeight, nonExpandableItemWeight }) => {
     const treeHead = { type: "list", children: [] };
-    const nodeWeight = { list: listWeight, openItem: openItemWeight, closedItem: closedItemWeight };
+    const nodeWeight = { list: listWeight, expandableItem: expandableItemWeight, nonExpandableItem: nonExpandableItemWeight };
 
-    // The head will be counted on each iteration.
-    let totalNodes = -MIN_NUMBER_OF_MAX_DEPTH_BRANCHES + 1;
+    let totalNodes = listWeight;
     for (let i = 0; i < MIN_NUMBER_OF_MAX_DEPTH_BRANCHES; i++) {
         let currentDepth = 0;
         let currentNode = treeHead;
         while (currentDepth < MAX_GENERATED_DOM_DEPTH) {
-            let type = (currentDepth + 1) % 2 ? "openItem" : "list";
-            currentNode.children.push({ type: type, children: [] });
+            let childType = (currentDepth + 1) % 2 ? "expandableItem" : "list";
+            currentNode.children.push({ type: childType, children: [] });
             currentNode = currentNode.children[currentNode.children.length - 1];
             currentDepth++;
-            totalNodes += nodeWeight[type];
+            totalNodes += nodeWeight[childType];
         }
-        if (MAX_GENERATED_DOM_DEPTH % 2) {
-            currentNode.type = "closedItem";
-            totalNodes = totalNodes - nodeWeight["openItem"] + nodeWeight["closedItem"];
+        // So these branches actually end in a todo list (nonExpandableItem) instead of a sprint folder.
```

Comment:
NIT: reword "Ensure the last node is not an expandableItem"

Reply:
Done.